app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -61,48 +61,30 @@ class SchoolDisciplineDashboard extends React.Component {
       title: "Incidents by " + selectedChart};
   }
 
-  sortChartData(chart) {
-    switch(chart.type) {
-      case 'time': return this.sortByTime(chart.disciplineIncidents);
-      case 'day': return this.sortByDay(chart.disciplineIncidents);
-      default: return chart.disciplineIncidents;
+  sortChartKeys(chartKeys) {
+    switch(this.state.selectedChart) {
+    case 'time': return this.sortedTimes(chartKeys);
+    case 'day': return this.sortedDays(chartKeys);
+    case 'grade': return this.sortedGrades(chartKeys);
+    default: return chartKeys;
     }
   }
 
-  sortByDay(incidentsGroupedByDay) {
-    const dayOrder = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
-    let sortedHash = {};
-    dayOrder.forEach((day) => {
-      if (incidentsGroupedByDay[day]) sortedHash[day] = incidentsGroupedByDay[day];
-    });
-    return sortedHash;
+  sortedDays(chartKeys) {
+    return ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'].filter((day) => chartKeys.indexOf(day) >= 0);
   }
 
-  sortByTime(incidentsGroupedByTime) {
-    const sortedTimes = Object.keys(incidentsGroupedByTime).sort((a, b) => {
-      // Times are coming in as '5 AM', '4 PM', or 'Not Logged'
-
-      if (a === b) return 0;
-      if (a === 'Not Logged') return -1;
-      if (b === 'Not Logged') return 1;
-
-      let a_Hour = parseInt(a.split(' ')[0]);
-      let a_AmPm = a.split(' ')[1];
-      let b_Hour = parseInt(b.split(' ')[0]);
-      let b_AmPm = b.split(' ')[1];
-
-      if (a_AmPm === 'AM' && b_AmPm == 'PM') return -1;
-
-      if (a_AmPm === 'PM' && b_AmPm == 'AM') return 1
-
-      return a_Hour - b_Hour;
+  sortedTimes(chartKeys) {
+    return chartKeys.sort((a, b) => {
+      if (a == "Not Logged") return -1;
+      if (b == "Not Logged") return 1;
+      return new Date('1970/01/01 ' + a) - new Date('1970/01/01 ' + b);
```

Comment:
If I understand this right, 1970/01/01 is an arbitrary value and is just here so we can compare two times as `new Date(...) - new Date(...)`, is that right?

Reply:
@alexsoble it does seem to work with just new Date(a) and it's probably fine to do that. My thinking was I'm not totally sure how it handles edge cases where a - b crosses into another day. Adding the time to a specific date removes ambiguity about which time is bigger.